### DISCUSSION TITLE
Problem with Redisson and Elasticache

### DISCUSSION THREAD

**Discussion by williamjvr** [vladimir-bukhtoyarov/bucket4j#422] [Q&A] [answered]
Hello everyone,

i configured my application to use redisson(3.23.3) and bucket4j(8.3.0) , my redis cluster is a elasticache at aws , but when my application start to receive a lot of requests, the application is down, because out of memory, my doubt is , Bucket4j support cluster with redisson or a i need to change to lettuce, you already seen this problem in another case 

- Spring boot Version -> 3.1.4
- Java Version -> 17
- redis url on properties ->   rediss://cluster.xxx.xxredis-cluster.xxx.sae1.cache.amazonaws.com:6379

**Comment by vladimir-bukhtoyarov** [comment, marked as answer]
What is request rate on single key? Currently all Redis integrations is built on top of Compare-and-swap paradigm. CAS should works good with high request rate when requests are well distributed across keys, and obviously CAS sucks for fat keys. If you have per user limits I suppose that it should works with accaptable latency, but if you have per country or single limit for all users it will not work without special settings from [this package](https://github.com/bucket4j/bucket4j/tree/master/bucket4j-core/src/main/java/io/github/bucket4j/distributed/proxy/optimization). And obviously, replacing Redisson by Lettuci will not help, because architecure is still the same. Currently,I [am awaiting](https://github.com/bucket4j/bucket4j/issues/410#issuecomment-1763919611) for next major Redis release `8.0`, which will provide Javascript functions, and it will helpfull to replace not scallable CAS by server-side procedures, in order to achive comparable latency with JSR-107 based integrations.

According to Redis Cluster and Redisson. This marked in docs as not supported
![image](https://github.com/bucket4j/bucket4j/assets/6761830/31221270-3369-4c51-a637-dd44eb6c8f38)
But really, I do not know if it can be supported or not. I just do not know how to configure Redis Cluster with Reddisson, and obviously I do not want to pay for Redisson license if this feature is not provided by Redisson for free.

**Reply by williamjvr** [reply]
Thank you for reply Vladimir, i change redisson to lettuce and gave a better performance and now my application is stable
